priv/stdlib/Bitwise.aes

```diff
@@ -0,0 +1,94 @@
+@compiler >= 4.3
+
+namespace Bitwise =
+
+  // bit shift 'x' right 'n' postions
+  function bsr(n : int, x : int) : int =
+    let step = 2^n
+    let res  = x / step
+    if (x >= 0 || x mod step == 0)
+      res
+    else
+      res - 1
+
+  // bit shift 'x' left 'n' positions
+  function bsl(n : int, x : int) : int =
+    x * 2^n
+
+  // bit shift 'x' left 'n' positions, limit at 'lim' bits
+  function bsli(n : int, x : int, lim : int) : int =
+    (x * 2^n) mod (2^lim)
+
+  // bitwise 'and' for arbitrary precision integers
+  function band(a : int, b : int) : int =
+      if (a >= 0 && b >= 0)
+        uband_(a, b, 1, 0)       // Small optimization...
+      elif (b >= 0)
+        ubnand(b, -1 - a)
+      elif (a >= 0)
+        ubnand(a, -1 - b)
+      else
+        -1 - ubor(-1 - a, -1 - b)
+
+  // bitwise 'or' for arbitrary precision integers
+  function
+    bor : (int, int) => int
+    bor(0, b) = b
+    bor(a, 0) = a
+    bor(a : int, b : int) : int =
+      if (a >= 0 && b >= 0)
+        ubor_(a, b, 1, 0)       // Small optimization...
+      elif (b >= 0)
+        -1 - ubnand(-1 - a, b)
+      elif (a >= 0)
+        -1 - ubnand(-1 - b, a)
+      else
+        -1 - uband(-1 - a, -1 - b)
+
+  // Bitwise 'and' for positive integers
```

Comment:
Since this is defined for `0`.

```suggestion
  // Bitwise 'and' for non-negative integers
```

Reply:
I got it right in the documentation though 🤦